src/averageAges.js

```diff
@@ -1,61 +1,61 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let filteredByCentury = people.filter(person =>
+    Math.ceil(person.died / 100) === century
+  );
+
+  if (century === undefined) {
+    filteredByCentury = people;
+  }
+
+  const filteredBySex = filteredByCentury.filter(person => person.sex === 'm');
+
+  const sumOfMenAges = filteredBySex.reduce((sum, men) =>
+    sum + (men.died - men.born), 0);
```

Comment:
DRY (Don't repeat yourself):
You use the same logic several times, it is better to move it to separate helper function and reuse it